### DISCUSSION TITLE
Don't enable a feature if there's a missing configuration

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#16953] [General]
As we know many OC features require configurations that come from different sources. I noticed many issues over the years occurred because of a bad or missing configuration. I'm suggesting not to enable a feature when there's no configuration unless the configuration comes from the UI only

Display an error message could be helpful too

**Comment by ns8482e** [comment]
Help me understand what missing configuration? When a feature is enabled and for example it has certain configuration needed then it should provide settings UI and user can set those configuration via UI

**Reply by hishamco** [reply]
For instance Email, Azure AD .. etc, we need to set the configuration through UI or configuration file, but some features require to be configured through `appsettings.json` such as Redis. So, when the feature is enabled it might be causes a runtime exception due missing configuration

**Reply by ns8482e** [reply]
In such case I feel that such  feature should be not available to be enabled/disabled via  feature UI, and should always be configure via code and app-settings
